recipes/libqasm/all/conanfile.py

```diff
@@ -0,0 +1,116 @@
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMakeDeps, CMake
+from conan.tools.files import copy, get
+from conan.tools.scm import Version
+
+
+class LibqasmConan(ConanFile):
+    name = "libqasm"
+
+    # Optional metadata
+    license = "Apache-2.0"
+    homepage = "https://github.com/QuTech-Delft/libqasm"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Library to parse cQASM files"
+    topics = ("code generation", "parser", "compiler", "quantum compilation", "quantum simulation")
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_python": [True, False],
+        "build_tests": [True, False],
+        "compat": [True, False],
+        "tree_gen_build_tests": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_python": False,
+        "build_tests": False,
+        "compat": False,
+        "tree_gen_build_tests": False
+    }
+
+    @property
+    def _min_cppstd(self):
+        return "20"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "13",
+            "clang": "13",
+            "gcc": "10",
+            "msvc": "192",
+            "Visual Studio": "16"
+        }
+
+    def build_requirements(self):
+        self.tool_requires("m4/1.4.19")
+        if self.settings.os == "Windows":
+            self.tool_requires("winflexbison/2.5.24")
+        else:
+            self.tool_requires("flex/2.6.4")
+            self.tool_requires("bison/3.8.2")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        self.folders.source = "."
+        self.folders.build = os.path.join("build", str(self.settings.build_type))
+        self.folders.generators = os.path.join(self.folders.build, "generators")
+
+        self.cpp.package.libs = ["cqasm"]
+        self.cpp.package.includedirs = ["include"]
+        self.cpp.package.libdirs = ["lib"]
+
+        self.cpp.source.includedirs = ["include"]
+        self.cpp.build.libdirs = ["."]
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+        tc = CMakeToolchain(self)
+        tc.variables["LIBQASM_BUILD_PYTHON"] = self.options.build_python
+        tc.variables["LIBQASM_BUILD_TESTS"] = self.options.build_tests
+        tc.variables["LIBQASM_COMPAT"] = self.options.compat
+        tc.variables["TREE_GEN_BUILD_TESTS"] = self.options.tree_gen_build_tests
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
```

Comment:
```suggestion
        # Fix broken hash for https://github.com/QuTech-Delft/tree-gen/tree/9ee414a7ade4b0ff293afeb095e6da0b2f44bb7a
        replace_in_file(self, os.path.join(self.source_folder, "src", "CMakeLists.txt"),
                        "c3d2d1a041a5826d3e377e4722774cbb8264d7c0",
                        "d708ea629bad014011d3f1efc352077414dd86f7")
        cmake = CMake(self)
```
The `tree-gen` git hash in the project is broken. Replaced it with the closest one in time to this version.

Reply:
We have now a `tree-gen/1.06` Conan package. We should use that instead.